compiler/ast/ast.go

```diff
@@ -292,6 +292,14 @@ type (
 		Args []Assignment `json:"args"`
 	}
 
+	// An AmbiguousExprList proc is a list of assignments whose parent proc
+	// is unknown: It could be a Filter, Summarize, or Put proc. This
+	// will be determined in the semantic phase.
+	AmbiguousExprList struct {
```

Comment:
Ambiguity is a property of context not syntactic structure.  Since this is the AST reflecting syntax, it seems better to just call something more reflective than this and that will help us understand what this is.  Since this lives at the operator level, maybe OpExpr?   Also, though the *List naming pattern is used elsewhere, I think *s pattern works better here... OpExprs instead of OpExprList instead of AmbiguousExprList.

Reply:
Yep, much better!